server/src/main/java/org/elasticsearch/repositories/RepositoryData.java

```diff
@@ -859,11 +853,25 @@ private static void parseSnapshots(
                     case VERSION:
                         version = Version.fromString(parser.text());
                         break;
+                    case START_TIME_MILLIS:
+                        assert startTimeMillis == -1;
+                        startTimeMillis = parser.longValue();
+                        break;
+                    case END_TIME_MILLIS:
+                        assert endTimeMillis == -1;
+                        endTimeMillis = parser.longValue();
+                        break;
+                    case PARTIAL_INDICES:
```

Comment:
This adds some redundant information to this already problematically large blob. If you run into an issue where a few data nodes drop out for a while and there were lots of indices on those (e.g. some warm nodes) then this would completely blow up the size of this blob. I think we could calculate this information from `IndexMetadataGenerations` instead by adding some virtual generation for partially snapshot indices and record the information that way without making the  blob that much larger in bad corner cases?

Reply:
I'll do the easier two fields first (start/end time) and then work on this separately.

Reply:
FWIW even in rather pathological cases I reckon this only inflates the blob by ~1MB or so per snapshot.

Reply:
Probably but going for a few hundred snapshots before someone realizes an issue isn't unheard of :) This has already been an issue with index meta generations before we started applying some tricks around deduplicating strings there.